src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -798,12 +709,20 @@ public int getCurrentPoolSize() {
      * @return current number of workers executing tasks
      */
     public int getCurrentRunningCount() {
-      synchronized (workersLock) {
-        int workerOutQty = currentPoolSize - availableWorkers.size();
-        if (waitingForWorker || currentPoolSize == 0) {
-          return workerOutQty;
+      int startPoolSize = currentPoolSize.get();
+      while (true) {
+        int idleWorkerCount = 0;
+        Worker idleWorker = this.idleWorker.get();
+        while (idleWorker != null) {
+          idleWorkerCount++;
+          idleWorker = idleWorker.nextIdleWorker;
+        }
+      
+        if (currentPoolSize.get() == startPoolSize) {
+          // TODO - do we want to re-check if the idle queue changed during computation
```

Comment:
honestly I think we should just get the idle count then return currentPoolSize.get() - idleWorkerCount.  No need to check/recheck anything.  I mean I get there is a chance we created a thread(s) that made it into idleWorkers while we did this operation but I dont see any reason we need to be 100% exact for that edge case.